kore/src/main/scala/org/kframework/compile/MergeRules.scala

```diff
@@ -30,6 +30,9 @@ class MergeRules(val automatonAttribute: String, filterAttribute: String) extend
   val isRulePredicate = KLabel("isRule")
 
   def apply(m: Module): Module = {
+    // TODO: find a better way to avoid adding merged rules to the spec module
+    if (m.claims.nonEmpty) return m
```

Comment:
@dwightguth I need your input on this.
Since adding all steps to the kprove pipeline, `MergeRules` is being called and it creates a rule which the backend doesn't expect. The most simple solution I could find was to avoid this step on modules that contain claims.